### ISSUE TITLE
Navigation issue for the whole app

### ISSUE THREAD

**Issue by Rajat-Sen** [StatCan/StatcanPublicationUAT#295]
Will be updated shortly!

**Comment by Rajat-Sen** [comment]
### Publications, Publication & For you screen

When the user goes to Publication screen to see an article and then goes to For your screen after following a subject, then taps on an article on the For you screen, first it shows the article from Publication screen and then loads the article from For you screen. Additionally, the tab bar changes from For you to publications. [The tab bar should be hidden on Publication screen as the task of the screen is to read article.]

After that, if the user presses the back button on Publication screen, opened from For you, it takes the user to For you screen. However, if the user slides back, it takes the user to Publications screen. [The behavior of the app should stay natural and take the user to For you screen after sliding back.]

To summarize, It should load a new screen to show a new article.

### Manage subjects, manage notifications and Search screen.

All these screens have temporary screens and they are not implemented properly.

If the user clicks on Manage subjects, manage notifications, and filter by subject, it should open up a temporary screen to make it feel natural and the user should remain on the same screen with the blurred background. The user should not be navigated to another screen.

**Comment by Jiandong-Guo** [comment]
Not sure what exactly to change, after reading the comment above, seems:
1) Article screen flashes(when an article is already shown in level 2 screen, user wants to show another article after navigating to another tab, the old article will flash for a second).   This is easy one. will fix it.
2)When displaying an article in level 2 screen no matter from Search or For you, The publication Tab will be always tinted, that is   because the level 2 screen is the child of publication Tab, we shared the same level2 screen everywhere.  if want to change it, we have to change the architecture, big change.
3)I guess the "temporary screen" you said is the actually you mean the pop-up dialog. 
      3a)  In Search screen, we do have pop-up dialog for Filter, when the modal dialog pops up, the surrounding area will become ugly blurred background by default, if you want to use default we can change it with easy.
      3b)The "Manage notifications", I guess you mean the tool bar menu item in Notification list screen, there is no "temporary screen" in Notification list screen, when this menu item is clicked, it will navigate to Manage Notification setting screen which is a child of setting tab, you will notice the tinted tab changes to setting.  this way we try to share as much code as possible.  if you want to add a  "temporary screen" inside the Notification list screen, that is fine even it's a big architecture change and against the "Coding once, Using everywhere" programming principle.  The setback is the app size and performance, because we'll have to keep 2 duplicated code sections in app bundle and run-time memory . 
    3c)The "Manage subjects", I guess you mean the tool bar menu item  in For you screen, there is no "temporary screen" in For you screen, when this menu item is clicked, it will navigate to Manage subjects setting screen which is a child of setting tab, you will notice the tinted tab changes to setting.  this way we try to share as much code as possible.  if you want to add a  "temporary screen" inside the For you screen, that is fine even it's a big architecture change and against "Coding once, Used everywhere" programming principle.  The setback is the app size and performance, because we'll have to keep 5 duplicated code sections in app bundle and run-time memory(one for Manage subjects Setting screen, one for For you screen, one for Following screen, one for Following list screen, one for Saved screen) . 

Please confirm

**Comment by zhuelle** [comment]
@lamokar10 Would you please provide an update on the ticket? Thanks!

**Comment by zhuelle** [comment]
@lamokar10 , this is a ticket with Major priority and in the current Spring. Since it is assigned to you and Rajat at the moment,  this is the reason that I am asking for an update from you. Should we wait for Rajat to proceed this ticket? Thanks!

**Comment by zhuelle** [comment]
In Today's virtual stand-up, the team discussed this ticket and we would like to proceed the ticket like this:

- For the issue described in Rajat previous comments about "Publications, Publication & For you screen", this is UX related and we will wait for Rajat coming back from his vacation to clarify the requirements and then proceed.
- For the issue described in Rajat previous comments about " Manage subjects, manage notifications and Search screen." @lamokar10 will compare it with the requirement and if it is an enhancement then she will create a change request ticket to ask the business to approve the enhancement.

**Comment by Rajat-Sen** [comment]
This ticket has been sub-divided into smaller tickets.
